types/module/configurator.go

```diff
@@ -38,3 +61,51 @@ func (c configurator) MsgServer() grpc.Server {
 func (c configurator) QueryServer() grpc.Server {
 	return c.queryServer
 }
+
+// RegisterMigration implements the Configurator.RegisterMigration method
+func (c configurator) RegisterMigration(moduleName string, forVersion uint64, handler MigrationHandler) error {
+	if forVersion == 0 {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidVersion, "module migration versions should start at 1")
+	}
+
+	if c.migrations[moduleName] == nil {
+		c.migrations[moduleName] = map[uint64]MigrationHandler{}
+	}
+
+	if c.migrations[moduleName][forVersion] != nil {
```

Comment:
I see there is a check for duplicates, but I don’t see any concurrency protection. Is that intended? If this code is to be executed under a lock, I’d highly suggest that we add the suffix “Locked” to the function name.

Reply:
How could there be any concurrency issues? It needs to get executed synchronously in the state machine

Reply:
I wasn’t sure of whether it needed any, which is why I asked :-) Thanks for the clarification. I shall collapse my comment to avoid distractions.

Reply:
Thanks 🙏